tests/v2/validation/charts/backup_restore/backup_restore.go

```diff
@@ -0,0 +1,321 @@
+
+	return createdSecret.Name, err
+}
+
+func createProject(client *rancher.Client, clusterID string) (*management.Project, error) {
+	createProject, _, err := projectsapi.CreateProjectAndNamespace(client, clusterID)
+	if err != nil {
+		return nil, err
+	}
+
+	return createProject, nil
+}
+
+func createRole(client *rancher.Client, context string, roleName string, rules []management.PolicyRule) (role *management.RoleTemplate, err error) {
+	role, err = client.Management.RoleTemplate.Create(
+		&management.RoleTemplate{
+			Context: context,
+			Name:    roleName,
+			Rules:   rules,
+		})
+
+	return
+}
+
+func createRancherResources(client *rancher.Client, resourceCount int, clusterID string) ([]*management.User, []*management.Project, []*management.RoleTemplate, error) {
+	userList := []*management.User{}
+	projList := []*management.Project{}
+	roleList := []*management.RoleTemplate{}
+
+	for i := 0; i < resourceCount; i++ {
+		u, err := users.CreateUserWithRole(client, users.UserConfig(), globalUserRole)
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		userList = append(userList, u)
+
+		p, err := createProject(client, clusterID)
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		projList = append(projList, p)
+
+		rt, err := createRole(client, clusterRoleContext, roleName, rules)
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		roleList = append(roleList, rt)
+	}
+
+	return userList, projList, roleList, nil
+}
+
+func createBackup(client *rancher.Client, name string) (*v1.SteveAPIObject, error) {
+	backup := setBackupObject()
+	backupTemplate := bv1.NewBackup("", name, backup)
+	completedBackup, err := client.Steve.SteveType(backupSteveType).Create(backupTemplate)
+
+	return completedBackup, err
+}
+
+func createRestore(client *rancher.Client, fileName string) (*v1.SteveAPIObject, error) {
+	restore := setRestoreObject()
+	restoreTemplate := bv1.NewRestore("", "", restore)
+	restoreTemplate.Spec.BackupFilename = fileName
+	completedRestore, err := client.Steve.SteveType(restoreSteveType).Create(restoreTemplate)
+
+	return completedRestore, err
+}
+
+func validateAWSS3BackupExists(bucket string, folder string, key string) (bool, error) {
+	if len(folder) > 0 {
+		folder += key
+		isPresent, err := checkAWSS3Object(bucket, folder)
+		return isPresent, err
+	}
+	isPresent, err := checkAWSS3Object(bucket, key)
+	return isPresent, err
+}
+
+func checkAWSS3Object(bucket string, key string) (bool, error) {
+	sdkConfig, err := awsConfig.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return false, err
+	}
+
+	s3Client := s3.NewFromConfig(sdkConfig)
+	_, err = s3Client.HeadObject(context.TODO(), &s3.HeadObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+	})
+
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case "NotFound":
+				return false, nil
+			default:
```

Comment:
A switch statement isn't necessary for 1 case; this could be covered in an if statement.

Reply:
I would also argue that this check should be done outside of the helper function, as not found can be an error, but it depends on the use case. But since this is a helper its up to your discretion.

Reply:
I grabbed this from an online resource and didn't think much of it since it worked 😅  makes sense though to do it as an if statement instead and I'll look into doing the check outside the helper function.